cluster-autoscaler/cloudprovider/gce/gce_manager_test.go

```diff
@@ -412,6 +413,44 @@ func TestGetMigSize(t *testing.T) {
 	size, err := g.GetMigSize(mig)
 	assert.NoError(t, err)
 	assert.Equal(t, int64(3), size)
+
+	size, err = g.GetMigSize(mig)
+	assert.Equal(t, int64(3), size)
+	mock.AssertExpectationsForObjects(t, server)
+}
+
+func TestGetAndSetMigSizeWithCache(t *testing.T) {
+	server := NewHttpServerMock()
+	defer server.Close()
+	g := newTestGceManager(t, server.URL, false)
+
+	server.On("handle", "/project1/zones/us-central1-b/instanceGroupManagers/extra-pool-323233232").Return(instanceGroupManagerResponseTemplate).Once()
+	server.On("handle", "/project1/zones/us-central1-b/instanceGroupManagers/extra-pool-323233232/resize").Return(setMigSizeResponse).Once()
+	server.On("handle", "/project1/zones/us-central1-b/operations/operation-1505739408819-5597646964339-eb839c88-28805931").Return(setMigSizeOperationResponse).Once()
+	server.On("handle", "/project1/zones/us-central1-b/instanceGroupManagers/extra-pool-323233232").Return(instanceGroupManagerResponseTemplate).Once()
+
+	mig := &gceMig{
+		gceRef: GceRef{
+			Project: projectId,
+			Zone:    zoneB,
+			Name:    "extra-pool-323233232",
+		},
+		gceManager: g,
+		minSize:    0,
+		maxSize:    1000,
+	}
+	size, err := g.GetMigSize(mig)
+	assert.NoError(t, err)
+	assert.Equal(t, int64(3), size)
+
+	err = g.SetMigSize(mig, 3)
```

Comment:
Can you use a different value? This would pass if SetMigSize was a no-op.

Reply:
Sure, changed it. Got lazy there and didn't add a response. Though, I don't think that it'll be a no-op.